tests/karma/core/containers/TestHandleLogin.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import { renderIntoDocument } from 'react-addons-test-utils';
+import { findDOMNode } from 'react-dom';
+import { Provider } from 'react-redux';
+import cookie from 'react-cookie';
+import { createStore } from 'redux';
+import HandleLogin, { mapDispatchToProps } from 'core/containers/HandleLogin';
+import * as api from 'core/api';
+
+describe('<HandleLogin />', () => {
+  class MyRouter extends React.Component {
+    static propTypes = {
+      children: React.PropTypes.node.isRequired,
+      router: React.PropTypes.object.isRequired,
+    }
+    static childContextTypes = {
+      router: React.PropTypes.object,
+    };
+
+    getChildContext() {
+      return {router: this.props.router};
+    }
+
+    render() {
+      return this.props.children;
+    }
+  }
+
+  function render(store, location, router) {
+    return findDOMNode(renderIntoDocument(
+      <Provider store={store}>
+        <MyRouter router={router}>
+          <HandleLogin location={location} />
+        </MyRouter>
+      </Provider>
+    ));
+  }
+
+  describe('while loading', () => {
+    const code = 'fxacode';
+    const state = 'statedata:base64path';
+    const location = {query: {code, state}};
+    const store = createStore((s = {}) => s, {api: {}, auth: {}});
+    let mockApi;
+    let router;
+
+    beforeEach(() => {
+      mockApi = sinon.mock(api);
+      mockApi.expects('login').withArgs({api: {}, code, state}).returns(Promise.resolve());
```

Comment:
I reckon new lines for each chained call (after  the first one) make it easier to read imho. It's a similar thing to newlines for descriptors in CSS, I suspect vertical line scanning is less cognitive overhead than horizontal `</anecData>`